user/serializers.py

```diff
@@ -1 +1,27 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "password", "email", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {
+            "password": {"write_only": True, "min_length": 5},
+        }
+
+    def create(self, validated_data):
+        """create user with encrypted password"""
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        """update user with encrypted password"""
+        password = validated_data.pop("password", None)
+        user = super().update(instance, validated_data)
+
+        if password:
+            user.set_password(password)
+            user.save()
```

Comment:
These lines are setting and saving the password only if it is provided in the update request. If it's not provided, the user's password could potentially be unset. Consider moving this logic to a place where you can ensure the password is always set, regardless of whether it's provided in the update request or not.